sycl/include/CL/sycl/detail/type_traits.hpp

```diff
@@ -32,6 +35,11 @@ using half = detail::half_impl::half;
 template <typename ElementType, access::address_space Space> class multi_ptr;
 
 namespace detail {
+// Type for Intel device UUID extension.
+// For details about this extension, see
+// sycl/doc/extensions/IntelGPU/IntelGPUDeviceInfo.md
+using uuid_type = std::array<std::byte, 16>;
```

Comment:
It seems `std::byte` is C++17 feature. Could you please either replace it with `unsigned char` or guard with C++17 check?

Reply:
This is an interesting point.  I think the declaration here is part of the public API for the extension.  The Khronos SYCL group has adopted C++17 as the base language for SYCL, so we use C++17 features when it makes sense to do so.  This does seem like a place where it makes sense.

Do DPC++ features follow the same principle where we use C++17 features where it makes sense?  If so, then I guess we would guard this code with an #ifdef, which means the feature will not be available for customers who try to compile in C++14 mode?

Reply:
Yes, the idea is to guard it so the feature is available if an app is compiled with C++17 or later.
Tagging @pvchupin

Reply:
if the UUID feature is available **only** when compiled with C++17 or later, don't we need the guard on others?

Reply:
> don't we need the guard on others?

Not sure what you mean.  Guard on other features?  This feature only needs the guard because it uses `std::byte` which is not available on compilers older than C++17.  Other features don't use `std::byte`.  I think there are a few other SYC features that use C++17 things, and those are also guarded.  (I'm not sure about this, though.)

Reply:
sorry for the confusion. I meant putting guard on the other code segs of this PR.

Reply:
I think the guard is only needed on code that exists in the public header files because those are compiled by the end users, and the end user might have an older C++ compiler.  We always compile DPC++ itself with C++17, so there is no need to guard code in the internals of the runtime because we know that compiler will always support C++17.

Reply:
the guard is causing the buildbot fails for c++14: http://ci.llvm.intel.com:8010/#/builders/37/builds/9166

Reply:
what is the expected behavior here? @romanovvlad

Reply:
It seems uses of `uuid_type` need also be guarded or we need to have an else branch with an alternative implementation of `uuid_type`.

Reply:
another option could be using `unsigned char` instead of `std::byte` as `else` branch would have it any way?

Reply:
we decided to go with `unsigned char`.